packages/apputils-extension/src/shortcuts.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+
+import { JupyterFrontEnd } from '@jupyterlab/application';
+import { CommandRegistry } from '@lumino/commands';
+import { Selector } from '@lumino/domutils';
+import * as React from 'react';
+import { Dialog, showDialog } from '@jupyterlab/apputils';
+import { TranslationBundle } from '@jupyterlab/translation';
+import '../style/base.css';
+
+export function displayShortcuts(
+  app: JupyterFrontEnd,
+  trans: TranslationBundle
+) {
+  const { commands } = app;
+  const elt = document.activeElement;
+
+  /**
+   * Find the distance from the target node to the first matching node.
+   *
+   * Based on Lumino private function commands.Private.targetDistance
+   * This traverses the DOM path from `elt` to the root
+   * computes the distance from `elt` to the first node which matches
+   * the CSS selector. If no match is found, `-1` is returned.
+   *
+   * It also stops traversal if the `data-lm-suppress-shortcuts` or
+   * `data-p-suppress-shortcuts` attributes are found.
+   */
+  function matchDistance(selector: string, elt: Element | null): number {
+    let targ = elt;
+    for (
+      let dist = 0;
+      targ !== null && targ !== targ.parentElement;
+      targ = targ.parentElement, ++dist
+    ) {
+      if (targ.hasAttribute('data-lm-suppress-shortcuts')) {
+        return -1;
+      }
+      if (targ.matches(selector)) {
+        return dist;
+      }
+    }
+    return -1;
+  }
+
+  function formatKeybinding(keys: readonly string[]): string {
+    return keys.map(CommandRegistry.formatKeystroke).join(', ');
+  }
+
```

Comment:
Since https://github.com/jupyterlab/lumino/pull/433, we should directly use Lumino method for consistency.

```suggestion
```